### PR TITLE
RATIS-1679. Shade io.perfmark:perfmark-api in ratis-thirdparty

### PR DESCRIPTION
**Pull request by codings-dan**
see https://issues.apache.org/jira/browse/RATIS-1679

### CODE REVIEW

**Comment by codings-dan** [comment]
@szetszwo Could you help take a look at this pull request? Thanks!

**Review by adoroszlai — APPROVED** [review]
Thanks @codings-dan for the fix.  Ran gRPC tests locally, worked fine.

File: pom.xml [hunk 1]
```diff
@@ -175,6 +178,11 @@
         <artifactId>metrics-ganglia</artifactId>
         <version>${shaded.dropwizard.ganglia.version}</version>
       </dependency>
+      <dependency>
+        <groupId>io.perfmark</groupId>
+        <artifactId>perfmark-api</artifactId>
+        <version>${shaded.perfmark.version}</version>
+      </dependency>
```
Comment (szetszwo) [inline pom.xml:185]: The io.perfmark dependency comes with gRPC so that we don't have add it here.

**Review by szetszwo — COMMENTED** [review]
@codings-dan , thanks for the update!  We only have to relocate io.perfmark but don't have to add the dependency.  The changes below are all what we need.
```xml
$git diff
diff --git a/misc/pom.xml b/misc/pom.xml
index eab42a9..7071684 100644
--- a/misc/pom.xml
+++ b/misc/pom.xml
@@ -190,6 +190,10 @@
                   <pattern>info.ganglia</pattern>
                   <shadedPattern>${ratis.thirdparty.shaded.prefix}.info.ganglia</shadedPattern>
                 </relocation>
+                <relocation>
+                  <pattern>io.perfmark</pattern>
+                  <shadedPattern>${ratis.thirdparty.shaded.prefix}.io.perfmark</shadedPattern>
+                </relocation>
               </relocations>
               <filters>
                 <!-- Filter unshaded netty_tcnative libs -->
@@ -239,7 +243,6 @@
                   <!-- Transitive stuff we don't need coming from google deps -->
                   <exclude>org.codehaus.mojo:animal-sniffer-annotations</exclude>
                   <exclude>org.slf4j:slf4j-api</exclude>
-                  <exclude>io.perfmark:perfmark-api</exclude>
                   <exclude>org.acplt:oncrpc</exclude>
                   <exclude>com.google.api.grpc:proto-google-common-protos</exclude>
                 </excludes>
```

File: pom.xml (hunk 1 above)
Comment (codings-dan) [reply pom.xml]: done

**Comment by codings-dan** [comment]
@szetszwo Thanks for helping review the pull request, I have updated it, please take a look. Also thanks @adoroszlai for helping review and test it!

**Review by szetszwo — APPROVED** [review]
+1 the change looks good.

**Comment by codings-dan** [comment]
@szetszwo Thanks for helping review the pull request!

**Pull request merged**